x86: Activate CONFIG_FB_DEVICE

hauke opened the pull request:
This adds the legacy /dev/fb* device file for kernel 6.6 again. Linux upstream commit 701d2054fa31 ("fbdev: Make support for userspace interfaces configurable") made this configurable and we deactivated this option by default for kernel 6.6. On x86 we are not space constrained and some users need this legacy interface.

Fixes: #15222

hauke committed:
x86: Activate CONFIG_FB_DEVICE

This adds the legacy /dev/fb* device file for kernel 6.6 again.
Linux upstream commit 701d2054fa31 ("fbdev: Make support for
userspace interfaces configurable") made this configurable and we
deactivated this option by default for kernel 6.6. On x86 we are not
space constrained and some users need this legacy interface.

Fixes: #15222
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

namiltd commented:
What about CONFIG_FB_VIRTUAL also mentioned in https://github.com/openwrt/openwrt/issues/15222?

hauke committed:
kernel: Add CONFIG_FB_DEVICE to kmod-fb

At least kmod-fb-tft depends on CONFIG_FB_DEVICE and can not be
activated without it.

This configuration option was added with kernel 6.6, before this featre
was always activated.

Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented:
`CONFIG_FB_VIRTUAL` was already available in older kernel versions and we did not activate it. This is a new option. I think we should do this separately. I do not plan to do it.
